lib/ocpp/v201/smart_charging.cpp

```diff
@@ -436,6 +511,55 @@ std::vector<ChargingProfile> SmartChargingHandler::get_profiles() const {
     return all_profiles;
 }
 
+ClearChargingProfileResponse SmartChargingHandler::clear_profiles(const ClearChargingProfileRequest& request) {
+    ClearChargingProfileResponse response;
+
+    const auto profile_id = request.chargingProfileId;
+    const auto criteria = request.chargingProfileCriteria;
+
+    auto any_found_profile = false;
+    // iterate over all profiles
+    for (auto& [existing_evse_id, evse_profiles] : charging_profiles) {
+        for (auto it = evse_profiles.begin(); it != evse_profiles.end();) {
+            // K10.FR.04: logical AND between the filters, if not set, the filter does not apply
+            if (profile_matches_clear_criteria(*it, profile_id, criteria, existing_evse_id)) {
+                any_found_profile = true;
+                evse_profiles.erase(it);
```

Comment:
I believe this line should be:

it = evse_profiles.erase(it);

Otherwise, it is an invalidated iterator and can't be depended upon to point to the correct part of the vector.